From 4f0995f7efa097f7f438b84053dd60075cf2ba06 Mon Sep 17 00:00:00 2001
From: Jason Jae Hyun Koh <jason.koh@motorola.com>
Date: Fri, 8 Aug 2014 12:34:06 -0500
Subject: [PATCH 064/959] IKLOCSEN-1580 misc:stml0xx: fix algo req data shifted
 by 1 byte

Algo requests were being corrupted due to this erroneous shift

Change-Id: Id43af962df52c89885b43aadf18b73e6290d4d83
Signed-off-by: Jason Jae Hyun Koh <jason.koh@motorola.com>
Reviewed-on: http://gerrit.mot.com/661263
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Robert Wiegand <Robert.P.Wiegand@motorola.com>
Reviewed-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/misc/stml0xx/stml0xx_ioctl.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/stml0xx/stml0xx_ioctl.c b/drivers/misc/stml0xx/stml0xx_ioctl.c
index 13e064e035e..1c4202a39d3 100644
--- a/drivers/misc/stml0xx/stml0xx_ioctl.c
+++ b/drivers/misc/stml0xx/stml0xx_ioctl.c
@@ -419,6 +419,7 @@ long stml0xx_misc_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		dev_dbg(&stml0xx_misc_data->spi->dev,
 			"Set algo req, algo idx: %d, len: %u", algo_idx, len);
 		if (algo_idx < STML0XX_NUM_ALGOS) {
+			buf[0] = stml0xx_algo_info[algo_idx].req_register;
 			dev_dbg(&stml0xx_misc_data->spi->dev, "Register: 0x%x",
 				stml0xx_algo_info[algo_idx].req_register);
 		} else {
@@ -433,7 +434,7 @@ long stml0xx_misc_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 			err = -EFAULT;
 			break;
 		}
-		if (copy_from_user(&buf[1], argp + 2 * sizeof(unsigned char)
+		if (copy_from_user(buf, argp + 2 * sizeof(unsigned char)
 				   + sizeof(len), len)) {
 			dev_err(&stml0xx_misc_data->spi->dev,
 				"Set algo req copy req info returned error");
@@ -441,7 +442,7 @@ long stml0xx_misc_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 			break;
 		}
 		stml0xx_g_algo_requst[algo_idx].size = len;
-		memcpy(stml0xx_g_algo_requst[algo_idx].data, &buf[1], len);
+		memcpy(stml0xx_g_algo_requst[algo_idx].data, buf, len);
 		err =
 		    stml0xx_spi_send_write_reg(stml0xx_algo_info
 					       [algo_idx].req_register, buf,
-- 
2.11.0

